### PR TITLE
Display program name when needed

### LINKED ISSUE #386: application: display Program name in addition to Round name
**User Story**
As a project owner, 
I want to know the name of the Program that is running the round I am applying to
so that I can confirm that I am applying to the correct round

**Acceptance Criteria**
- Make sure the program name is being displayed on **Round Application page**
- Make sure the program name is being displayed on **Apply to a Round banner**
- Make sure the program name is being displayed on **Application Page**

Note: Round Manager team has now added program contract address to round metadata, so we should be able to pull in Program name programatically (discussion: https://discord.com/channels/562828676480237578/973967228577935400/1017772055858528307) 

**Designs**

**Round Application Landing Page**
Screen Shot 2022-09-23 at 9 21 49 AM

**Apply to a Round banner**
Screen Shot 2022-09-23 at 9 22 19 AM

**Application Page**
Screen Shot 2022-09-23 at 9 22 39 AM

### PR DESCRIPTION
**Pull request by DanieleSalatti**
Addressing #386 

~~WIP, tests should break right now~~

### CODE REVIEW

File: client/src/__tests__/components/grants/List.test.tsx [hunk 1]
```diff
@@ -213,15 +212,28 @@ describe("<List />", () => {
...
           store.dispatch({ type: "ROUND_APPLICATION_NOT_FOUND", roundAddress });
 
           renderWrapped(<List />, store);
 
-          expect(
-            screen.getByText(
-              "Apply to Optimism Grant Round and get your project funded!"
-            )
-          ).toBeInTheDocument();
+          expect(screen.getByText("Apply to Round")).toBeInTheDocument();
```
Comment (gravityblast) [inline client/src/__tests__/components/grants/List.test.tsx:236]: shall we check `Apply to {round.programName}` so we know we are using the program name?

File: client/src/__tests__/components/grants/List.test.tsx [hunk 2]
```diff
@@ -233,11 +245,7 @@ describe("<List />", () => {
 
           renderWrapped(<List />, store);
 
-          expect(
-            screen.queryByText(
-              "Apply to Optimism Grant Round and get your project funded!"
-            )
-          ).toBeNull();
+          expect(screen.queryByText("Apply to Round")).toBeNull();
```
Comment (gravityblast) [inline client/src/__tests__/components/grants/List.test.tsx:248]: same as above

**Review by gravityblast — APPROVED** [review]
LGTM! I just added a couple of comments

**Comment by codenamejason** [comment]
@DanieleSalatti is this still WIP? I need this PR to merge to do my next task.

**Pull request merged**